File: ontrack-extension-general/src/main/java/net/nemerosa/ontrack/extension/general/BuildLinkDecoration.java
package net.nemerosa.ontrack.extension.general;

import lombok.Data;
import net.nemerosa.ontrack.model.structure.PromotionRun;

import java.net.URI;
import java.util.List;

@Data
public class BuildLinkDecoration {

    private final String project;
    private final String build;
    private final URI uri;
    private final List<PromotionRun> promotionRuns;

}

File: ontrack-extension-general/src/main/java/net/nemerosa/ontrack/extension/general/BuildLinkDecoration.kt
package net.nemerosa.ontrack.extension.general

/**
 * List of build links, plus an indicator (link to the build page)
 * showing if there is extra links.
 *
 * @property buildId ID of the parent link
 * @property linksCount Number of downstream links
 */
class BuildLinkDecorationList(
        val buildId: Int, // Needed for NextUI
        val linksCount: Int, // Needed for NextUI
)
